libshaderc/src/shaderc_test.cc

```diff
@@ -956,4 +956,48 @@ TEST_F(CompileKindsTest, TessEvaluation) {
                                  shaderc_glsl_tess_evaluation_shader));
 }
 
+// A test case for ParseVersionProfileTest needs: 1) the input string, 2)
+// expected parsing results, including 'succeed' flag, version value, and
+// profile enum.
+struct ParseVersionProfileTestCase {
+  std::string str;
+  bool expected_succeed;
+  int expected_version;
+  shaderc_profile expected_profile;
+};
+
+// Test for a helper function to parse version and profile from string.
+using ParseVersionProfileTest =
+    testing::TestWithParam<ParseVersionProfileTestCase>;
+
+TEST_P(ParseVersionProfileTest, FromNullTerminatedString) {
+  const ParseVersionProfileTestCase& test_case = GetParam();
+  int version;
```

Comment:
good practice to initialize scalar non-class values. (version and profile in this case.)
Some compiler somewhere complain about later uses without initializations.  Just avoid that so we don't convert that warning to an error.